src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmApiServiceImpl.java

```diff
@@ -71,13 +74,30 @@ public DownloadResponseMetaData downloadArmData(String externalRecordId, String
     private String getArmBearerToken() {
         log.debug("Get ARM Bearer Token with Username: {}, Password: {}", armApiConfigurationProperties.getArmUsername(),
                   armApiConfigurationProperties.getArmPassword());
+        String accessToken = null;
         ArmTokenResponse armTokenResponse = armTokenClient.getToken(new ArmTokenRequest(
             armApiConfigurationProperties.getArmUsername(),
             armApiConfigurationProperties.getArmPassword(),
             GrantType.PASSWORD.getValue()
         ));
+
+        if (StringUtils.isNotEmpty(armTokenResponse.getAccessToken())) {
+            AvailableEntitlementProfile availableEntitlementProfile = armTokenClient.availableEntitlementProfiles(armTokenResponse.getAccessToken());
+            if (!availableEntitlementProfile.isError()) {
+                Optional<String> profileId = availableEntitlementProfile.getProfiles().stream()
+                    .filter(p -> armApiConfigurationProperties.getArmServiceProfile().equalsIgnoreCase(p.getProfileName()))
+                    .map(p -> p.getProfileId())
+                    .findFirst();
```

Comment:
findFirst means that java will traverse the list in-order, but a findAny will mean it searches it in any order but stops after it finds one, so findAny may be more performant.
Also, not sure if it would make any difference, but should the findFirst/findAny come before the map?

Reply:
Changed findFirst to findAny. It doesnt have to come before the map.